### DISCUSSION TITLE
Excel Import

### DISCUSSION THREAD

**Discussion by aldids01** [filamentphp/filament#13673] [Help] [answered]
### Package

Actions

### Package Version

v3.2

### How can we help you?

<?php

namespace App\Filament\Imports;

use App\Models\Category;
use Filament\Actions\Imports\ImportColumn;
use Filament\Actions\Imports\Importer;
use Filament\Actions\Imports\Models\Import;

class CategoryImporter extends Importer
{
    protected static ?string $model = Category::class;

    public static function getColumns(): array
    {
        return [
            ImportColumn::make('name')
                ->requiredMapping()
                ->rules(['required', 'max:255']),
        ];
    }

    public function resolveRecord(): ?Category
    {
//         return Category::firstOrNew([
//             // Update existing records, matching them by `$this->data['column_name']`
//             'name' => $this->data['name'],
//         ]);

        return new Category();
    }

    public static function getCompletedNotificationBody(Import $import): string
    {
        $body = 'Your category import has completed and ' . number_format($import->successful_rows) . ' ' . str('row')->plural($import->successful_rows) . ' imported.';

        if ($failedRowsCount = $import->getFailedRowsCount()) {
            $body .= ' ' . number_format($failedRowsCount) . ' ' . str('row')->plural($failedRowsCount) . ' failed to import.';
        }

        return $body;
    }
}

the above is my importer created in app\filament\imports the importation is successful but its not sending the data to database what am i doing wrong?

**Comment by leandrocfe** [comment]
`php artisan queue:work`
?

**Reply by aldids01** [reply]
Thank you when i run the php artisan queue:work, the import and export are now working and the notifications are been sent to the database, but still what happen when i depoloy it to a local server, is there a way to handle the queue without necessary runing php artisan queue:work?

![image](https://github.com/user-attachments/assets/f2a9e47d-33f7-4e72-b6b9-d4fb2b724737)
![image](https://github.com/user-attachments/assets/f3e87865-5e1b-4fbe-9c18-65b24c5c4e42)

**Reply by leandrocfe** [reply, marked as answer]
If you don't use queues, set `QUEUE_CONNECTION=sync` in your .env

**Reply by aldids01** [reply]
Okay Thanks

**Reply by aldids01** [reply]
Please Help me out with this last question, I want to display my table record based on a condition, apart of using filter, i can't find it in the documentation. For example, i want to display employee records where status = 1, i don't want to use filter, and tab to do that, how can i achieved it?
